From cf86f2f2f640dd2880c41c73e2b90afbbf2abdb9 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Thu, 4 May 2017 02:01:56 +0200
Subject: [PATCH] Use ".../src" as temporary package build directory

instead of <package name>-<version>. This makes build path
predictably shorter.

Closes: #860554
---
 runner/autopkgtest | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/runner/autopkgtest b/runner/autopkgtest
index 2f43446..fe7e93d 100755
--- a/runner/autopkgtest
+++ b/runner/autopkgtest
@@ -321,7 +321,7 @@ def build_source(kind, arg, built_binaries):
             p = adt_testbed.Path(testbed, part, os.path.join(testbed.scratch, os.path.basename(part)))
             p.copydown()
 
-        create_command = 'dpkg-source -x "%s"' % dsc_tb
+        create_command = 'dpkg-source -x "%s" src' % dsc_tb
 
     elif kind == 'unbuilt-tree':
         dsc = os.path.join(tmp, 'fake.dsc')
@@ -385,10 +385,10 @@ def build_source(kind, arg, built_binaries):
                           '  ver=${srcversion:-$pkg_candidate};'
                           '  dpkg --compare-versions "$ver" lt "$maxver" || maxver="$ver";'
                           'done;'
-                          '[ -z "$maxver" ] || maxver="=$maxver";'
-                          'OUT=$(apt-get source -q --only-source %(src)s$maxver 2>&1) || RC=$?;'
+                          '[ -z "$maxver" ] || maxver="$maxver";'
+                          'OUT=$(apt-get source -d -q --only-source %(src)s=$maxver 2>&1) || RC=$?;'
                           'if [ -n "$RC" ]; then if echo "$OUT" | grep -q "Unable to find a source package"; then exit 1; else exit $RC; fi; fi;'
-                          'echo "$OUT" | grep ^Get: || true' % {'src': arg})
+                          'echo "$OUT" | grep ^Get: ; dpkg-source -x %(src)s_$maxver.dsc src' % {'src': arg})
     elif kind == 'git-source':
         url, _, branch = arg.partition('#')
         create_command = "git clone '%s' || { sleep 15; git clone '%s'; }" % (url, url)
-- 
2.7.4

